src/cljs/commiteth/manage_payouts.cljs

```diff
@@ -1,91 +1,356 @@
+   [:span.usd-balance-label (str "$" value-usd)]])
+
+(defn token-balances [crypto-balances]
+  [:span ; TODO consider non DOM el react wrapping
+   (for [[tla balance] crypto-balances]
+     ^{:key tla}
+     [:div.dib.mr2
+      [balance-badge tla balance]])])
+
+(defn bounty-balance [{:keys [value-usd] :as bounty}]
+  [:div
+   [token-balances (bnt/crypto-balances bounty)]
+   [:div.dib.mr2.pv1
+    [usd-value-label value-usd]]])
+
+(defn bounty-card [{owner        :repo-owner
+                    repo-name    :repo-name
+                    issue-title  :issue-title
+                    issue-number :issue-number
+                    updated      :updated
+                    tokens       :tokens
+                    balance-eth  :balance-eth
+                    value-usd    :value-usd
+                    :as bounty}
+                   {:keys [style] :as opts}]
+  [:div
+   [:a {:href (bnt/issue-url bounty)}
+    [:div.cf
+     [:div.fl.w-80
+      [:span.db.f4.muted-blue.hover-black issue-title]
+      #_[:div.mt2
+       [:span.f5.gray.pg-book (str owner "/" repo-name " #" issue-number)]]]
+     [:div.fl.w-20.tr
+      [:span.f6.gray.pg-book
+       {:on-click #(do (.preventDefault %) (prn (dissoc bounty :claims)))}
+       (common/human-time updated)]]]]])
+
+(defn confirm-button [bounty claim]
+  (let [paid?   (bnt/paid? claim)
+        merged? (bnt/merged? claim)]
+    (when (and merged? (not paid?))
+      [:button.f7.ttu.tracked.outline-0.bg-sob-blue.white.pv3.ph4.pg-med.br3.bn.pointer
+       (merge {:on-click #(rf/dispatch [:confirm-payout claim])}
+              (if (and merged? (not paid?) (:payout_address bounty))
+                {}
+                {:disabled true})
+              (when (and (or (bnt/confirming? bounty)
+                             (bnt/bot-confirm-unmined? bounty))
+                         merged?)
+                {:class "busy loading" :disabled true}))
+       (if paid?
+         "Signed off"
+         "Confirm Payment")])))
+
+(defn confirm-row [bounty claim]
+  (let [payout-address-available? (:payout_address bounty)]
+    [:div
+     (when-not payout-address-available?
+       [:div.bg-sob-blue-o-20.pv2.ph3.br3.mb3.f6
+        [:p [:span.pg-med (or (:user_name claim) (:user_login claim))
+             "’s payment address is pending."] " You will be able to confirm the payment once the address is provided."]])
+     [:div.cf
+      [:div.dt.fr
+       (when-not payout-address-available?
+         {:style {:-webkit-filter "grayscale(1)"
+                  :pointer-events "none"}})
+       [:div.dtc.v-mid.pr3.f6
+        [bounty-balance bounty]]
+       [:div.dtc.v-mid
+        [confirm-button bounty claim]]]]]))
+
+(defn claim-card [bounty claim {:keys [render-view-claim-button?] :as opts}]
+  (let [{user-name :user_name
+         user-login :user_login
+         avatar-url :user_avatar_url} claim
+        winner-login (:winner_login bounty)]
+    [:div.pa2
+     [:div.flex.items-center
```

Comment:
not sure which of these is really the right line to comment on. but long term does it make sense to have a central "card" component which is reused by bounty & claim card, or are the structural differences with each too great for that to be of much value?

Reply:
It could make sense to have that card, but in my experience, you often need some minor variation and so such component would need to be highly configurable.